.circleci/config.yml

```
@@ -0,0 +1,24 @@
+version: 2.1
+jobs: 
+  build: 
+    docker: 
+      - image: rust:1
+    steps:
+      - checkout
+      - run:
+          name: Verify Clean Code
+          command: |
+            rustup toolchain install stable-x86_64-unknown-linux-gnu
+            rustup component add clippy
+            cargo clippy
+      - run:
+          name: Verify Proper Formatting
+          command: |
+            rustup component add rustfmt
+            cargo fmt -- --check
+      - run:
+          name: Verify Code Builds
+          command: cargo build --all --release
+      - run:
+          name: Verify Tests Pass
+          command: cargo test --all --release
```

https://doc.rust-lang.org/cargo/commands/cargo-build.html#package-selection
> `--all`
Deprecated alias for `--workspace`.

Doesn't change any functionality, but should probably change to `--workspace`
```suggestion
          command: cargo build --workspace --release
      - run:
          name: Verify Tests Pass
          command: cargo test --workspace --release
```